scripts/zones/West_Ronfaure/npcs/Vilatroire.lua

```diff
@@ -5,29 +5,222 @@
+                    if (member:getNation() == player:getNation()) then
+                        partySameNationCount = partySameNationCount + 1;
+                    end
+
+                    -- check race for second
+                    if (player:getRace() == tpz.race.HUME_M and member:getRace() == tpz.race.HUME_M or member:getRace() == tpz.race.HUME_F) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.HUME_F and member:getRace() == tpz.race.HUME_M or member:getRace() == tpz.race.HUME_F) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.ELVAAN_M and member:getRace() == tpz.race.ELVAAN_M or member:getRace() == tpz.race.ELVAAN_F) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.ELVAAN_F and member:getRace() == tpz.race.ELVAAN_M or member:getRace() == tpz.race.ELVAAN_F) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.TARU_M and member:getRace() == tpz.race.TARU_M or member:getRace() == tpz.race.TARU_F) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.TARU_F and member:getRace() == tpz.race.TARU_M or member:getRace() == tpz.race.TARU_F) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.GALKA and member:getRace() == tpz.race.GALKA) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    elseif (player:getRace() == tpz.race.MITHRA and member:getRace() == tpz.race.MITHRA) then
+                        partySameRaceCount = partySameRaceCount + 1;
+                    end
+
+                    -- job for third
+                    if (member:getMainJob() == player:getMainJob()) then
+                        partySameJobCount = partySameJobCount + 1;
+                    end
+                end
+            end
+
+            if (questIntroToTeamwork == QUEST_ACCEPTED) then
+                -- https://www.bg-wiki.com/bg/Introduction_to_Teamwork
+                if (partySameNationCount == partySizeRequirement) then
+                    -- nation requirements met
+                    player:setCharVar("introToTmwrk_pass", 1);
+                    player:updateEvent(15, 1);
+                else
+                    -- not met
+                    player:updateEvent(3);
+                end
+            elseif (questIntermediateTeamwork == QUEST_ACCEPTED) then
+                -- https://www.bg-wiki.com/bg/Intermediate_Teamwork
+                if (partySameRaceCount == partySizeRequirement) then
+                    -- race requirements met
+                    player:setCharVar("intermedTmwrk_pass", 1);
```

Comment:
These can be local vars, no need to set in database as the delete in event finish